lib/library.rb

```diff
@@ -0,0 +1,51 @@
+class Library
+  attr_reader :name,
+              :books,
+              :authors,
+              :checked_out_books,
+              :checkout_tally
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @checkout_tally = Hash.new(0)
+  end
+
+  def add_author(author)
+    @authors.push(author)
+
+    author.books.each do |book|
+      @books.push(book)
+    end
+  end
+
+  def publication_time_frame_for(author)
+    publication_years = author.books.map do |book|
+      book.publication_year.to_i
+    end
+
+    { start: publication_years.min.to_s,
+      end:   publication_years.max.to_s }
+  end
```

Comment:
this works well. I'd probably look at min_by and max_by to do this and avoid the map call, since you're effectively iterating over your data three times: once for `.map`, once for `.min` and once for `.max`